### PR TITLE
[Classic UI] Consider item options for selection widget when no mapping is defined

### PR DESCRIPTION
**Pull request by lolodomo**
Relative to issue #4942

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by lolodomo** [comment]
@cweitkamp : can you please review and approve this one too ? It is the same principle as for Basic UI.

File: extensions/ui/org.eclipse.smarthome.ui.classic/src/main/java/org/eclipse/smarthome/ui/classic/internal/render/SelectionRenderer.java [hunk 1]
```diff
@@ -69,40 +72,16 @@ public boolean canRender(Widget w) {
...
-            } else {
-                rowSnippet = StringUtils.replace(rowSnippet, "%checked%", "");
+        } else {
+            for (Mapping mapping : selection.getMappings()) {
+                String rowSnippet = buildRow(selection, mapping.getLabel(), mapping.getCmd(), item, state);
+                rowSB.append(rowSnippet);
             }
-            rowSB.append(rowSnippet);
         }
         snippet = StringUtils.replace(snippet, "%label_header%", getLabel(w, mappedValue));
```
Comment (cweitkamp) [inline extensions/ui/org.eclipse.smarthome.ui.classic/src/main/java/org/eclipse/smarthome/ui/classic/internal/render/SelectionRenderer.java]: Only a question for my personal understanding: What is the difference between `getLabel(w, mappedValue)` and `getLabel(w)`?
See https://github.com/eclipse/smarthome/pull/5667/files#diff-955d3a6399e421b68151ddb7351de8bbR107

File: extensions/ui/org.eclipse.smarthome.ui.classic/src/main/java/org/eclipse/smarthome/ui/classic/internal/render/SelectionRenderer.java [hunk 2]
```diff
@@ -114,6 +93,42 @@ public boolean canRender(Widget w) {
...
+
+        State compareMappingState = state;
+        if (state instanceof QuantityType) { // convert the item state to the command value for proper
+                                             // comparison and "checked" attribute calculation
+            compareMappingState = convertStateToLabelUnit((QuantityType<?>) state, command);
+        }
+
+        if (compareMappingState.toString().equals(command)) {
+            rowSnippet = StringUtils.replace(rowSnippet, "%checked%", "checked=\"true\"");
+            mappedValue = (label != null) ? label : command;
```
Comment (cweitkamp) [inline extensions/ui/org.eclipse.smarthome.ui.classic/src/main/java/org/eclipse/smarthome/ui/classic/internal/render/SelectionRenderer.java]: Why do we assign the `command` here if the `label` is `null`? We don't check for that in Basic UI.
See https://github.com/eclipse/smarthome/pull/5667/files#diff-955d3a6399e421b68151ddb7351de8bbR144

File: extensions/ui/org.eclipse.smarthome.ui.classic/src/main/java/org/eclipse/smarthome/ui/classic/internal/render/SelectionRenderer.java [hunk 3]
```diff
@@ -44,6 +45,8 @@
 
     private final Logger logger = LoggerFactory.getLogger(SelectionRenderer.class);
 
+    private String mappedValue;
```
Comment (kaikreuzer) [inline extensions/ui/org.eclipse.smarthome.ui.classic/src/main/java/org/eclipse/smarthome/ui/classic/internal/render/SelectionRenderer.java]: Do you really want that as an instance field? Isn't that an issue if there are concurrent requests that render different widgets with mappings at the very same moment?

File: extensions/ui/org.eclipse.smarthome.ui.classic/src/main/java/org/eclipse/smarthome/ui/classic/internal/render/SelectionRenderer.java (hunk 2 above)
Comment (lolodomo) [reply extensions/ui/org.eclipse.smarthome.ui.classic/src/main/java/org/eclipse/smarthome/ui/classic/internal/render/SelectionRenderer.java]: A label for a StateOption can be null, not its value.
For the mappings in the sitemap, the syntax forces you to define a label and a value.
So to be clear, this test was useless before my change but is now required.
So it is now ok here but I have to change Basic UI.

File: extensions/ui/org.eclipse.smarthome.ui.classic/src/main/java/org/eclipse/smarthome/ui/classic/internal/render/SelectionRenderer.java (hunk 3 above)
Comment (lolodomo) [reply extensions/ui/org.eclipse.smarthome.ui.classic/src/main/java/org/eclipse/smarthome/ui/classic/internal/render/SelectionRenderer.java]: That was an easy way as I have now a method that should return 2 data ... but not clean.
I will try something else.

**Comment by lolodomo** [comment]
New try without mappedValue as instance field.

**Pull request merged**